src/lib.rs

```diff
@@ -17,46 +17,151 @@
+            })
+            .with_mapped_port(Port {
+                local: wallet_rpc_port,
+                internal: 28083,
+            })
+            .with_entrypoint("")
+            .with_args(vec![
+                "/bin/bash".to_string(),
+                "-c".to_string(),
+                "monerod --confirm-external-bind --non-interactive --regtest --rpc-bind-ip 0.0.0.0 --rpc-bind-port 28081 --no-igd --hide-my-port --fixed-difficulty 1 --rpc-payment-allow-free-loopback --data-dir /monero & \
+                monero-wallet-rpc --log-level 4 --daemon-address localhost:28081 --confirm-external-bind --disable-rpc-login --rpc-bind-ip 0.0.0.0 --rpc-bind-port 28083  --wallet-dir /monero/".to_string(),
+            ])
+            .with_wait_for(WaitFor::LogMessage { message: "You are now synchronized with the network. You may now start monero-wallet-cli".to_string(), stream: Stream::StdOut });
+        let container = docker_client.run(image);
         Self {
-            wallet: wallet::Client::localhost(MONERO_WALLET_RPC_PORT)
+            container,
+            wallet: wallet::Client::localhost(wallet_rpc_port)
                 .expect("failed to create wallet client"),
-            monerod: monerod::Client::localhost(MONEROD_RPC_PORT)
+            monerod: monerod::Client::localhost(monerod_rpc_port)
                 .expect("failed to create monerod client"),
         }
     }
+
+    /// Constructor that generates random port numbers for the local port
+    /// mapping of `monerod` and `monero-wallet-rpc`.
+    pub fn new_with_random_local_ports(cli: &'c clients::Cli) -> Self {
+        let mut rng = rand::thread_rng();
+        let monerod_port: u16 = rng.gen_range(1024, u16::MAX);
+        let wallet_port: u16 = rng.gen_range(1024, u16::MAX);
+
+        Client::new(cli, monerod_port, wallet_port)
+    }
+
+    /// Initialise by creating a wallet, generating some `blocks`, and starting
+    /// a miner thread that mines to the primary account.
+    pub async fn init(&self, blocks: u32) -> Result<()> {
+        self.wallet.create_wallet("miner_wallet").await?;
+        let miner = self.wallet.get_address_primary().await?.address;
+
+        let _ = self.monerod.generate_blocks(blocks, miner.clone()).await?;
+
+        let _ = tokio::spawn(mine(self.monerod.clone(), miner));
+
+        Ok(())
+    }
+
+    /// Initialise by creating a wallet, generating some `blocks`, and starting
+    /// a miner thread that mines to the primary account. Also create two
+    /// sub-accounts, one for Alice and one for Bob. Fund Alice's account.
+    pub async fn init_with_accounts(&self) -> Result<()> {
+        self.wallet.create_wallet("miner_wallet").await?;
+        let alice = self.wallet.create_account("alice").await?;
+        let _bob = self.wallet.create_account("bob").await?;
+
+        let miner = self.wallet.get_address_primary().await?.address;
+
+        let res = self.monerod.generate_blocks(70, miner.clone()).await?;
```

Comment:
I'd add a note why 70 blocks. According to the specs it should be 60 but it seems only to work with 70 blocks.
```suggestion
		// 70 blocks seems to be the sweet number to have enough spendable funding.
        let res = self.monerod.generate_blocks(70, miner.clone()).await?;
```